internal/storage/db/internal/db.go

```diff
@@ -464,6 +478,55 @@ func (s *dbStorage) GetDependents(ctx context.Context, ids ...namer.ModuleID) (m
 	return out, nil
 }
 
+func (s *dbStorage) Disable(ctx context.Context, policyKey ...string) (int64, error) {
+	if len(policyKey) == 1 {
+		mID := namer.GenModuleIDFromFQN(namer.FQNFromPolicyKey(policyKey[0]))
+		res, err := s.db.Update(PolicyTbl).Prepared(true).
+			Set(goqu.Record{PolicyTblDisabledCol: true}).
+			Where(goqu.C(PolicyTblIDCol).Eq(mID)).
+			Executor().ExecContext(ctx)
+		if err != nil {
+			return 0, err
+		}
+
+		affected, err := res.RowsAffected()
+		if err != nil {
+			return 0, fmt.Errorf("failed to discover whether the policy got disabled or not: %w", err)
+		} else if affected == 0 {
+			return 0, fmt.Errorf("failed to find the policy for disabling")
+		}
+
+		s.NotifySubscribers(storage.NewPolicyEvent(storage.EventAddOrUpdatePolicy, mID))
+
+		return affected, nil
+	}
+
+	mIDList := make([]any, len(policyKey))
+	events := make([]storage.Event, len(policyKey))
+	for i, pk := range policyKey {
+		mID := namer.GenModuleIDFromFQN(namer.FQNFromPolicyKey(pk))
+		mIDList[i] = mID
+		events[i] = storage.Event{Kind: storage.EventAddOrUpdatePolicy, PolicyID: mID}
+	}
+	res, err := s.db.Update(PolicyTbl).Prepared(true).
+		Set(goqu.Record{PolicyTblDisabledCol: true}).
+		Where(goqu.C(PolicyTblIDCol).In(mIDList...)).
+		Executor().ExecContext(ctx)
+	if err != nil {
+		return 0, err
+	}
+
+	affected, err := res.RowsAffected()
+	if err != nil {
+		return 0, fmt.Errorf("failed to discover whether the policies got disabled or not: %w", err)
+	} else if affected == 0 {
+		return 0, fmt.Errorf("failed to find the policies for disabling")
+	}
```

Comment:
I don't think this is an error. If you issue the same call twice by accident, the second call would always return 0 rows.